cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -195,15 +195,15 @@ private boolean initAdminClient() {
 
         Promise<Void> result = Promise.promise();
         singleExecutor.submit(() -> {
-            List<Integer> podIds = new ArrayList<>(numPods);
+            List<Integer> podIds = new ArrayList<>(podList.size());
 
-            for (int podId = 0; podId < numPods; podId++) {
+            for (int podId = 0; podId < podList.size(); podId++) {
                 // Order the podIds unready first otherwise repeated reconciliations might each restart a pod
                 // only for it not to become ready and thus drive the cluster to a worse state.
-                podIds.add(podOperations.isReady(namespace, podName(podId)) ? podIds.size() : 0, podId);
+                podIds.add(podOperations.isReady(namespace, podList.get(podId)) ? podIds.size() : 0, podId);
             }
             LOGGER.debugCr(reconciliation, "Initial order for rolling restart {}", podIds);
-            List<Future> futures = new ArrayList<>(numPods);
+            List<Future> futures = new ArrayList<>(podList.size());
             for (Integer podId: podIds) {
                 futures.add(schedule(podId, 0, TimeUnit.MILLISECONDS));
```

Comment:
Does this really work? I do not see you changing the schedule method => so how will it get the pod name from the pod ID?